extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/websockets/WebSocketMetricsInterceptorProducerImpl.java

```diff
@@ -0,0 +1,137 @@
+package io.quarkus.micrometer.runtime.binder.websockets;
+
+import static io.quarkus.micrometer.runtime.binder.websockets.WebSocketMetricConstants.DIRECTION_TAG_KEY;
+import static io.quarkus.micrometer.runtime.binder.websockets.WebSocketMetricConstants.Direction.INBOUND;
+import static io.quarkus.micrometer.runtime.binder.websockets.WebSocketMetricConstants.Direction.OUTBOUND;
+
+import jakarta.enterprise.context.Dependent;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.quarkus.websockets.next.runtime.spi.telemetry.WebSocketMetricsInterceptorProducer;
+
+@Dependent
+public final class WebSocketMetricsInterceptorProducerImpl implements WebSocketMetricsInterceptorProducer {
+
+    private static final String URI_TAG_KEY = "uri";
```

Comment:
If we have path parameters, it can lead to an explosion. We need to apply the same approach as for HTTP.

Reply:
> If we have path parameters, it can lead to an explosion. We need to apply the same approach as for HTTP.

I answered this later in Bruno's thread, this is a route, not path parameter. I renamed arguments to make things clearer.